packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -564,10 +564,14 @@ class _FocusState extends State<Focus> {
 
   @override
   void dispose() {
+    if (!focusNode.hasFocus && widget.onFocusChange != null) {
+      widget.onFocusChange?.call(false);
+    }
+
     // Regardless of the node owner, we need to remove it from the tree and stop
     // listening to it.
-    focusNode.removeListener(_handleFocusChanged);
     _focusAttachment!.detach();
+    focusNode.removeListener(_handleFocusChanged);
```

Comment:
Thanks for updating the setup! Unfortunately it looks like many checks are still failing:

![some checks were not successful](https://github.com/user-attachments/assets/579a7858-2078-4c26-8bfb-a3609683a105)

<br>

Triggering the `onFocusChange` callback as the Focus is being disposed of could be a massively breaking change, since I bet there are many existing callbacks inside & outside the framework that involve `setState` calls.

My preference would be to close this PR and recommend that a developer supply a focus node if this sort of behavior is desired.

Reply:
I agree, let's close this PR for now. It was a productive discussion!

How should we respond in the issue section? Should I leave a comment to explain the decision?

Reply:
> I agree, let's close this PR for now. It was a productive discussion!

Love this energy!

<br>

> How should we respond in the issue section?

I already went ahead and left a comment there: https://github.com/flutter/flutter/issues/156285#issuecomment-2417956193
We can probably just wait and see how they choose to respond :)